call.go

```diff
@@ -213,10 +213,19 @@ func invoke(ctx context.Context, method string, args, reply interface{}, cc *Cli
 			// RPC has completed or failed.
 			put func()
 		)
+		codec := c.codec
+		if codec == nil {
+			codec = cc.dopts.codec
+		}
+		contentSubtype := c.contentSubtype
+		if contentSubtype == "" {
+			contentSubtype = codec.String()
+		}
 		// TODO(zhaoq): Need a formal spec of fail-fast.
 		callHdr := &transport.CallHdr{
-			Host:   cc.authority,
-			Method: method,
+			Host:           cc.authority,
+			Method:         method,
+			ContentSubtype: contentSubtype,
```

Comment:
Um. I do think we should keep the existing behaviour of using `application/grpc` if the user hasn't specified `proto` as their codec explicitly. 

Otherwise other user's code may silently break and it's unclear how it would interact with other gRPC implementations.